ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/GhcrIntegrationForm.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { ReactElement } from 'react';
+import { Flex, FlexItem, TextInput, PageSection, Form, Checkbox } from '@patternfly/react-core';
+import * as yup from 'yup';
+import merge from 'lodash/merge';
+
+import { ImageIntegrationBase } from 'services/ImageIntegrationsService';
+
+import usePageState from 'Containers/Integrations/hooks/usePageState';
+import ExternalLink from 'Components/PatternFly/IconText/ExternalLink';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormTestButton from 'Components/PatternFly/FormTestButton';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import IntegrationHelpIcon from './Components/IntegrationHelpIcon';
+import useIntegrationForm from '../useIntegrationForm';
+import { IntegrationFormProps } from '../integrationFormTypes';
+
+import IntegrationFormActions from '../IntegrationFormActions';
+import FormLabelGroup from '../FormLabelGroup';
+
+export type GhcrIntegration = {
+    categories: 'REGISTRY'[];
+    docker: {
+        endpoint: string;
+        username: string;
+        password: string;
```

Comment:
from what I can tell, it requires the use of a personal access token (classic) instead of a username and password: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry

has this been tested to see if a username/password works?

Reply:
FYI - [the docs show an example](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry:~:text=Using%20the%20CLI%20for%20your%20container%20type%2C%20sign%20in%20to%20the%20Container%20registry%20service%20at%20ghcr.io.) using `docker login` with `username` = GH username and `password` = personal access token.

So it _should_ work, but ➕ to testing regardless.

Reply:
I wonder if we can make it clear in the UI that this should be a personal access token (classic) instead of a password

Reply:
Do you suggest renaming `Password` to something like `Personal Access Token`?

Reply:
I renamed the UI field to `GitHub token`. For the record, the normal GitHub user password does not work, it has to be a token.